tests/infer/mcmc/test_mcmc_api.py

```diff
@@ -73,19 +73,24 @@ def normal_normal_model(data):
 @pytest.mark.parametrize('num_draws', [None, 1800, 2200])
 @pytest.mark.parametrize('group_by_chain', [False, True])
 @pytest.mark.parametrize('num_chains', [1, 2])
+@pytest.mark.filterwarnings("ignore:num_chains")
 def test_mcmc_interface(num_draws, group_by_chain, num_chains):
     num_samples = 2000
     data = torch.tensor([1.0])
-    initial_params, _, transforms, _ = initialize_model(normal_normal_model, model_args=(data,))
+    initial_params, _, transforms, _ = initialize_model(normal_normal_model, model_args=(data,),
+                                                        num_chains=num_chains)
     kernel = PriorKernel(normal_normal_model)
-    mcmc = MCMC(kernel=kernel, num_samples=num_samples, warmup_steps=100,
-                initial_params=initial_params, transforms=transforms)
+    mcmc = MCMC(kernel=kernel, num_samples=num_samples, warmup_steps=100, num_chains=num_chains,
+                mp_context="spawn", initial_params=initial_params, transforms=transforms)
     mcmc.run(data)
     samples = mcmc.get_samples(num_draws, group_by_chain=group_by_chain)
     # test sample shape
     expected_samples = num_draws if num_draws is not None else num_samples
     if group_by_chain:
         expected_shape = (mcmc.num_chains, expected_samples, 1)
+    elif num_draws is not None:
+        # FIXME: what is the expected behavior of num_draw is not None and group_by_chain=False?
+        expected_shape = (expected_samples, 1)
```

Comment:
I think this is an issue. We should address it in a separate PR to make the behavior consistent. Probably we can remove `num_samples` arg in `get_samples` and defer the resampling job to a utility.

Reply:
I think it is okay to draw `num_samples` from the combined pool if the samples are not grouped by chain. I think it will be even more surprising to return twice or thrice the number of samples unless the chain_dim is preserved. Alternatively we could rename this argument to imply number of samples per chain instead. Deprecation is a possibility too, but this shouldn't affect anything else so I'm fine keeping it.

Reply:
Yeah, I'm fine with it too.